accepted/install-locations.md

```diff
@@ -0,0 +1,227 @@
+
+There is not installer for this scenario, these installs are produced by simply copying the right directory structure into a custom location.
+
+*This mechanism already exists in .NET Core 2.2.*
+
+If the muxer (`dotnet`) is used then it must be invoked directly from the private install location (it's not registered on `PATH` in any way).  
+If other hosts are used then the private install location must be specified in environment variable:
+* `DOTNET_ROOT`
+* `DOTNET_ROOT(x86)` for 32-bit process running on 64-bit Windows.
+
+
+## Host behavior for 3.0
+
+### `hostfxr` search
+`dotnet` only looks right next to itself.
+
+All other hosts will search for the first location which exists in this order:
+* the directory where the host resides (this is important mostly for `apphost` as self-contained apps will have the `hostfxr` in the same directory).
+* `DOTNET_ROOT` (or `DOTNET_ROOT(x86)`) environment variable, in the `host/fxr/<version>` subdirectory.
+* :new: per-user registered install location, in the `/host/fxr/<version>` subdirectory
+  * Windows - `HKCU\SOFTWARE\dotnet\Setup\InstalledVersions\<arch>\InstallLocation`
+  * Linux/macOS - `$HOME/.dotnet/install_location`
+* :new: global registered install location, in the `/host/fxr/<version>` subdirectory  
+  **We've committed to support this on Windows for .NET Core 3.0**
+  * Windows - `HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\<arch>\InstallLocation` (32-bit view)  
+  * Linux/macOS - `/etc/dotnet/install_location`  
+* default global location, in the `/host/fxr/<version>` subdirectory
+  * Windows (32-bit process on 64-bit Windows) - `%ProgramFiles(x86)%\dotnet`
+  * Windows (everything else) - `%ProgramFiles%\dotnet`
+  * Linux - `/usr/share/dotnet`
+  * macOS - `/usr/local/share/dotnet`
+
+In each case with the `<version>` subdirectory, the latest available version as per SemVer 2.0 rules will be selected. Note that the algorithm picks the first location which exists, and only then looks for the `/host/fxr/<version>` subdirectory, and if that doesn't exist, or the library is missing it will fail.
+
+### Framework and SDK search
+All hosts use the same logic.
+
+Both framework and SDK search uses the same logic, the only difference is that frameworks are in the `share` subdirectory, while SDKs are in the `sdk` subdirectory of the install location.
+In both cases all locations listed below are searched and the complete list of all available SDKs/Frameworks is combined before making a choice which one will be used (based on version requirements).
+
+Search locations:
+* the location of the `hostfxr` used. See the `hostfxr` search algorithm above. This means that `DOTNET_ROOT` has effect on framework and SDK search as well.
+* If multi-level lookup is enabled (by default it is, can be disabled via `DOTNET_MULTILEVEL_LOOKUP=0` environment variable)
+  * :new: per-user registered install location
```

Comment:
This may be a security problem: It is not generally ok to run global "admin" installed programs against user-installed runtime because of it allows users to alter the behavior.

Reply:
Thanks for pointing that out - I didn't realize it. Based on other comments above as well as this one, I'll remove the per-user install from the document.